### PR TITLE
Update sql doc

### PR DESCRIPTION
**Pull request by Jefffrey**
# Which issue does this PR close?

Related to https://github.com/apache/arrow-datafusion/issues/3058

# Rationale for this change

Update sql doc and do some touching up

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/expr/src/aggregate_function.rs [hunk 1]
```diff
@@ -77,30 +77,34 @@ impl FromStr for AggregateFunction {
     type Err = DataFusionError;
     fn from_str(name: &str) -> Result<AggregateFunction> {
         Ok(match name {
-            "min" => AggregateFunction::Min,
-            "max" => AggregateFunction::Max,
-            "count" => AggregateFunction::Count,
+            // general
```
Comment (Jefffrey) [inline datafusion/expr/src/aggregate_function.rs:80]: just reorganizing, to be similar to how it looks in the sql doc, and group them together here a bit better

File: datafusion/expr/src/built_in_function.rs [hunk 2]
```diff
@@ -313,12 +313,13 @@ impl FromStr for BuiltinScalarFunction {
             // math functions
             "abs" => BuiltinScalarFunction::Abs,
             "acos" => BuiltinScalarFunction::Acos,
-            "asin" => BuiltinScalarFunction::Asin,
-            "atan" => BuiltinScalarFunction::Atan,
             "acosh" => BuiltinScalarFunction::Acosh,
+            "asin" => BuiltinScalarFunction::Asin,
```
Comment (Jefffrey) [inline datafusion/expr/src/built_in_function.rs:317]: ditto

File: docs/source/user-guide/sql/select.md [hunk 3]
```diff
@@ -153,6 +153,20 @@ either side of the join where there is not a match.
 +----------+----------+----------+----------+
 ```
 
+### NATURAL JOIN
```
Comment (comphead) [inline docs/source/user-guide/sql/select.md:156]: +1

**Review by comphead — APPROVED** [review]
Thanks for PR @Jefffrey.
@alamb I think going fwd we need to check doc update as part of review, to keep documentation in sync.
To be honest I cant' even imagine how hard it was to sync the code and documentation!

**Comment by comphead** [comment]
Thanks for PR @Jefffrey.
@alamb I think going fwd we need to check doc update as part of review, to keep documentation in sync.
To be honest I cant' even imagine how hard it was to sync the code and documentation!

**Comment by alamb** [comment]
> @alamb I think going fwd we need to check doc update as part of review, to keep documentation in sync.
> To be honest I cant' even imagine how hard it was to sync the code and documentation!

Yes I agree -- thank you @Jefffrey 

I don't have a good story for keeping the docs in sync. I will try and remember to verify the docs / ask for changes, but if we rely on a human process we'll likely always be behind -- automation is our best hope I think. 

Here is one idea to keep the function reference more in sync: https://github.com/apache/arrow-datafusion/issues/3092

There probably needs to be more too

File: docs/source/user-guide/sql/ddl.md [hunk 4]
```diff
@@ -19,6 +19,32 @@
...
+Create schema under specified catalog, or the default DataFusion catalog if not specified.
+
+<pre>
+CREATE SCHEMA [ IF NOT EXISTS ] [ <i><b>catalog.</i></b> ] <b><i>schema_name</i></b>
+</pre>
+
+```sql
+-- create schema emu under catalog cat
+CREATE SCHEMA cat.emu;
+```
```
Comment (alamb) [inline docs/source/user-guide/sql/ddl.md:46]: I didn't know this worked ❤️  but I tried it out 

```
8 rows in set. Query took 0.014 seconds.
❯ create table cat.emu.dog(x int);
0 rows in set. Query took 0.000 seconds.
❯ show tables;
+---------------+--------------------+-------------+------------+
| table_catalog | table_schema       | table_name  | table_type |
+---------------+--------------------+-------------+------------+
| cat           | emu                | dog         | BASE TABLE |
| cat           | information_schema | tables      | VIEW       |
| cat           | information_schema | views       | VIEW       |
| cat           | information_schema | columns     | VIEW       |
| cat           | information_schema | df_settings | VIEW       |
| datafusion    | information_schema | tables      | VIEW       |
| datafusion    | information_schema | views       | VIEW       |
| datafusion    | information_schema | columns     | VIEW       |
| datafusion    | information_schema | df_settings | VIEW       |
+---------------+--------------------+-------------+------------+
9 rows in set. Query took 0.001 seconds.
```

File: docs/source/user-guide/sql/select.md [hunk 5]
```diff
@@ -83,7 +83,7 @@ SELECT a FROM table WHERE a > 10
 
 ## JOIN clause
 
-DataFusion supports `INNER JOIN`, `LEFT OUTER JOIN`, `RIGHT OUTER JOIN`, `FULL OUTER JOIN`, and `CROSS JOIN`.
+DataFusion supports `INNER JOIN`, `LEFT OUTER JOIN`, `RIGHT OUTER JOIN`, `FULL OUTER JOIN`, `NATURAL JOIN` and `CROSS JOIN`.
```
Comment (alamb) [inline docs/source/user-guide/sql/select.md:86]: We also support SEMI JOIN and ANTI Join (can be documented in some other PR)

```
❯ create table foo(x int) as values (1);
0 rows in set. Query took 0.009 seconds.
❯ create table bar(x int) as values (1);
0 rows in set. Query took 0.001 seconds.
❯ select foo.x from foo left semi join bar using (x);
+---+
| x |
+---+
| 1 |
```

**Review by alamb — APPROVED** [review]
Thanks @Jefffrey  -- very nice 👏

**Pull request merged**